core/src/main/java/nl/nn/adapterframework/pipes/StringIteratorPipe.java

```diff
@@ -88,34 +89,34 @@ public void startBlock() throws SenderException, TimeOutException, IOException {
 				}
 			}
 			
-			private boolean finalizeBlock() throws SenderException, TimeOutException, IOException {
+			private StopReason finalizeBlock() throws SenderException, TimeOutException, IOException {
 				if (processingInBlocks && isCombineBlocks() && itemCounter>0) {
 					itemCounter=0;
 					items.append(getBlockSuffix());
-					boolean result = super.handleItem(items.toString());
+					stopReason = super.handleItem(items.toString());
 					items.setLength(0);
-					return result;
+					return stopReason;
 				}
-				return false;
+				return null;
 			}
 			
 			@Override
-			public boolean handleItem(String item) throws SenderException, TimeOutException, IOException {
+			public StopReason handleItem(String item) throws SenderException, TimeOutException, IOException {
 				if (processInBlocksBySize && itemCounter==0) {
 					startBlock();
 				} 
 				if (processInBlocksByKey) {
 					String key = getKey(item);
 					if (!key.equals(previousKey)) { 
-						if (previousKey!=null && !finalizeBlock()) {
-							return false;
+						if(finalizeBlock() != null) {
+							return stopReason;
```

Comment:
I would expect that the stopReason returned would have been the result of `finalizeBlock()`